rotkehlchen/chain/ethereum/modules/yearn/graph.py

```diff
@@ -0,0 +1,424 @@
+
+            try:
+                from_asset = EthereumToken(entry['vault']['token']['id'])
+                to_asset = EthereumToken(entry['vault']['shareToken']['id'])
+            except UnknownAsset:
+                from_str = entry['vault']['token']['symbol']
+                to_str = entry['vault']['shareToken']['symbol']
+
+                self.msg_aggregator.add_warning(
+                    f'Ignoring deposit in yearn V2 from {from_str} to '
+                    f'{to_str} because the token is not recognized.',
+                )
+                continue
+
+            # since the query of prices is expensive we check if
+            # it's in the dict and if not query it
+            from_asset_usd_price = prices_cache.get(from_asset)
+            to_asset_usd_price = prices_cache.get(to_asset)
+
+            if from_asset_usd_price is None:
+                prices_cache[from_asset] = get_usd_price_zero_if_error(
+                    asset=from_asset,
+                    time=Timestamp(int(entry['timestamp']) // 1000),
+                    location='yearn v2 vault deposit',
+                    msg_aggregator=self.msg_aggregator,
+                )
+                from_asset_usd_price = prices_cache.get(from_asset)
+
+            if to_asset_usd_price is None:
+                prices_cache[to_asset] = get_usd_price_zero_if_error(
+                    asset=to_asset,
+                    time=Timestamp(int(entry['timestamp']) // 1000),
+                    location='yearn v2 vault deposit',
+                    msg_aggregator=self.msg_aggregator,
+                )
+                to_asset_usd_price = prices_cache.get(to_asset)
+
+            assert to_asset_usd_price is not None
+            assert from_asset_usd_price is not None
+
+            from_asset_amount = token_normalized_value(
+                token_amount=int(entry['tokenAmount']),
+                token=from_asset,
+            )
+
+            to_asset_amount = token_normalized_value(
+                token_amount=int(entry['sharesMinted']),
+                token=to_asset,
+            )
+
+            result.append(YearnVaultEvent(
+                event_type='deposit',
+                block_number=int(entry['blockNumber']),
+                timestamp=Timestamp(int(entry['timestamp']) // 1000),
+                from_asset=from_asset,
+                from_value=Balance(
+                    amount=from_asset_amount,
+                    usd_value=from_asset_amount * from_asset_usd_price,
+                ),
+                to_asset=to_asset,
+                to_value=Balance(
+                    amount=to_asset_amount,
+                    usd_value=to_asset_amount * to_asset_usd_price,
+                ),
+                realized_pnl=None,
+                tx_hash=tx_hash,
+                log_index=int(log_index),
+            ))
+        return result
+
+    def _process_withdrawals(self, withdrawals: List[Any]) -> List[YearnVaultEvent]:
```

Comment:
For the entire function. Same comments as for the deposits one.

In fact the code is almost copy paste. Which means you should really really abstract it away into one common single function